src/Arcus.Templates.Tests.Integration/Worker/MessagePump/MessagePumpService.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Threading.Tasks;
+using Arcus.EventGrid;
+using Arcus.EventGrid.Parsers;
+using Arcus.EventGrid.Testing.Infrastructure.Hosts.ServiceBus;
+using Arcus.Messaging.ServiceBus.Core.Extensions;
+using Arcus.Templates.Tests.Integration.Fixture;
+using Arcus.Templates.Tests.Integration.Logging;
+using Arcus.Templates.Tests.Integration.Worker.Fixture;
+using Bogus;
+using GuardNet;
+using Microsoft.Azure.EventGrid.Models;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.ServiceBus.Core;
+using Microsoft.Extensions.Configuration;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Arcus.Templates.Tests.Integration.Worker.MessagePump
+{
+    /// <summary>
+    /// Represents a service to interact with the hosted-service.
+    /// </summary>
+    public class MessagePumpService : IAsyncDisposable
```

Comment:
Now you've lost me 😅 

What are we doing here and why do we need it? 😄

Reply:
Ah, here is some interesting logic bundled. This service allows us to interact with the message pump (that is part of a newly created project from the template created during the test).
What it does is sending a message on Service Bus and look if the currently runned project (from the template) picks this up.

In short, (like the descrpition says); it's a way to know that the project created from the template correctly runs the message pump, and that the message pump can process messages.

Reply:
Can't you just use the same approach as in the messaging testing where you create a pump implementation, push to grid and very in the test?

You are somewhat doing this but more complicated, although it might be because it's a template but not sure.

Reply:
Yeah, that's maybe the weird part. What you describe, is exactly what happens; only on another level (in the created project) and not in the test code. The `OrdersMessagePump.cs` file be added to the created project so it handles the Service Bus messages and place them on Event Grid.

Reply:
So, in a way, we're using the same approach.

Reply:
But how are we testing it if it's running in the test project? 🤔 

Can't we move it in the created project to simplify the whole structure like it would be in reality?

Reply:
1. [Test code] creates Service Bus message
2. [Test code] places Service Bus message on queue
3. [Demo project] processes Service Bus message on queue
4. [Demo project] places message on Event Grid topic
5. [Test code] subscribes on Event Grid topic and reads original message.

I don't think it would make much sense to move either the sending or receiving part, since it would make it even more obscure.

Reply:
Ok makes sense now, thanks!